src/main/java/com/vladyslavpalamarchuk/trackmycoin/command/CommandProcessorRegistry.java

```diff
@@ -12,22 +13,39 @@
 public class CommandProcessorRegistry {
 
   private final Map<Command, CommandProcessor> commandToProcessors;
+  private final Map<Long, Command> userLastCommands = new ConcurrentHashMap<>();
 
   public CommandProcessorRegistry(List<CommandProcessor> processors) {
     commandToProcessors =
         processors.stream()
-            .collect(Collectors.toMap(CommandProcessor::getCommand, Function.identity()));
+            .collect(
+                Collectors.toMap(
+                    com.vladyslavpalamarchuk.trackmycoin.command.processor.CommandProcessor
+                        ::getCommand,
+                    Function.identity()));
   }
 
-  public CommandProcessor get(String messageText) {
+  public CommandProcessor get(Long chatId, String messageText) {
```

Comment:
This method should be as simple as possible. Please move logic about adding monitor to separate command processor. You can think about sharing map with user command across all command processors and then each of them will have all needed info to do any logic that we need